File: Archived_Code_20200906_1604/Backup/Code_17122019_Backup_Do_Not_Delete/java-revisited/src/Sorting/BubbleSort.java
package Sorting;

import java.util.Arrays;

public class BubbleSort {

    public static void bubble_sort(int[] array) {
        for (int i = 0; i < array.length - 1; i++) {
            for (int j = 0; j < array.length - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int[] array = {14, 20, 24, 11, 16};
        System.out.println("Before sorting = " + Arrays.toString(array));
        bubble_sort(array);
        System.out.println("After sorting = " + Arrays.toString(array));
    }
}

File: Archived_Code_20200906_1604/Backup/Code_17122019_Backup_Do_Not_Delete/mount-everest/Sorting/bubble-sort.cc
#include <iostream>
#include <ctime>

void print(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        std::cout << arr[i] << " ";
    }
    std::cout << "\n";
}

void bubble_sort(int arr[], int n) {
    int temp;
    bool sorted = false;
    for (int k = 0; k < n-1; k++) {
        sorted = true;
        for (int i =0; i < n-k-1; i++) {
            if (arr[i] > arr[i+1]) {
                temp = arr[i];
                arr[i] = arr[i+1];
                arr[i+1] = temp;
                sorted = false;
            }
        }
        if (sorted) break;
    }
}

int main (int argc, char *argv[]) {
    srand(time(nullptr));
    const int min = 100;
    const int max = 999;
    const int size = 50;
    int arr[size] = {0};

    for (int i = 0; i < 50; i++) {
        arr[i] = min + rand() % (max - min + 1);
    }

    std::cout << "Without sorting:\n";
    print(arr, size);

    bubble_sort(arr, size);
    
    std::cout << "After sorting:\n";
    print(arr, size);

    return 0;
}

